packit_service/worker/events/event.py

```diff
@@ -465,7 +465,51 @@ def get_package_config(self) -> Optional[PackageConfig]:
         return package_config
 
     @staticmethod
-    def _filter_failed_models_targets(
+    def _get_submitted_time_from_model(
+        model: Union[CoprBuildTargetModel, TFTTestRunTargetModel]
+    ) -> datetime:
+        # TODO: unify `submitted_name` (or better -> create for both models `task_accepted_time`)
+        # to delete this mess plz
+        if isinstance(model, CoprBuildTargetModel):
+            return model.build_submitted_time
+
+        return model.submitted_time
+
+    @classmethod
+    def get_most_recent_targets(
+        cls,
+        models: Union[
+            Optional[Iterable[CoprBuildTargetModel]],
+            Optional[Iterable[TFTTestRunTargetModel]],
+        ],
+    ) -> Dict[str, Union[CoprBuildTargetModel, TFTTestRunTargetModel]]:
+        """
+        Gets most recent models from an iterable (regarding submission time).
+
+        Args:
+            models: Copr or TF models - if there are any duplicates in them then use the most
+             recent model
+
+        Returns:
+            Dictionary - target as a key and corresponding most recent model as a value.
+        """
+        most_recent_models: Dict[
+            str, Union[CoprBuildTargetModel, TFTTestRunTargetModel]
+        ] = {}
+        for model in models:
+            submitted_time_of_current_model = cls._get_submitted_time_from_model(model)
+            if (
+                most_recent_models.get(model.target) is None
+                or cls._get_submitted_time_from_model(most_recent_models[model.target])
+                < submitted_time_of_current_model
+            ):
+                most_recent_models[model.target] = model
+
+        return most_recent_models
```

Comment:
is there a reason to not return only the most recent target models (values)? Is/will this method be reused somewhere?

Reply:
I hope the method will be reused in the future for similar purposes.  But you are right, returning only values should be enough.